### PR TITLE
[Fix] Support more generic tenant code when create tenant

### LINKED ISSUE #9633: [Bug] [TenantManage/CreateTenant] Create a tenant is prompting an error message
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Create a tenant is prompting an error message

### What you expected to happen

create tenant  successful

### How to reproduce

my OS username is shimin.an

step:
1. create a tenant
2. input tenant name  shimin.an
3. click submit

### Anything else

image

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SnowMoon-Dev**
… error

## Purpose of the pull request

[Fix-9633][TenantManage/CreateTenant] Create a tenant is prompting an error

## Brief change log

close #9633 

## Verify this pull request

Manually verified the change by testing locally

### CODE REVIEW

**Comment by zhongjiajie** [comment]
could you add some test on it?

**Comment by SnowMoon-Dev** [comment]
there are already test cases, so I didn't add
@zhongjiajie

**Comment by zhongjiajie** [comment]
> there are already test cases, so I didn't add @zhongjiajie

I do not think so, if you change the rule, you should change the case too. In this case, you should add at least one test case with username contain char `.`

**Comment by SnowMoon-Dev** [comment]
i agree with you，i will add testcase

**Comment by SnowMoon-Dev** [comment]
@zhongjiajie I have added test cases

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/utils/RegexUtilsTest.java [hunk 1]
```diff
@@ -28,10 +28,10 @@ public class RegexUtilsTest {
     @Test
     public void testIsValidLinuxUserName() {
         String name1 = "10000";
-        Assert.assertFalse(RegexUtils.isValidLinuxUserName(name1));
+        Assert.assertTrue(RegexUtils.isValidLinuxUserName(name1));
```
Comment (zhongjiajie) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/utils/RegexUtilsTest.java:31]: I find out that you change de previous logic, and it support the user name start with number now?

**Review by zhongjiajie — APPROVED** [review]

**Pull request merged**